File: Onani/models/schemas/news.py
# -*- coding: utf-8 -*-
# @Author: kapsikkum
# @Date:   2022-03-09 20:03:24
# @Last Modified by:   kapsikkum
# @Last Modified time: 2022-03-09 20:10:47
from Onani.models import NewsPost

from . import ma


class NewsPostSchema(ma.SQLAlchemyAutoSchema):
    author = ma.auto_field()

    class Meta:
        model = NewsPost

File: Onani/static/js/modules/news.js
/**
 * @Author: kapsikkum
 * @Date:   2022-04-19 15:17:46
 * @Last Modified by:   kapsikkum
 * @Last Modified time: 2022-04-19 15:25:36
 */
import { parse as twemojiParse } from "twemoji";
import { DateTime } from "luxon";

const newsContainer = document.getElementById("news-container");

fetch("/api/news", { method: "GET" }).then((response) => {
  response.json().then((json) => {
    json.data.forEach((news) => {
      // Make a list entry element
      let newsPost = document.createElement("li"),
        newsLink = document.createElement("a"),
        newsTime = document.createElement("h6");

      // Set the href
      newsLink.href = `/news/${news.id}`;

      // set the title
      newsLink.innerText = news.title;

      // set the content
      newsTime.textContent = `(${DateTime.fromISO(
        news.created_at
      ).toRelativeCalendar()})`;

      // make it twemoji
      twemojiParse(newsLink);

      // append items
      newsPost.appendChild(newsLink);
      newsPost.appendChild(newsTime);
      newsContainer.appendChild(newsPost);
    });
  });
});
